lib/Jsrt/Jsrt.cpp

```diff
@@ -2511,6 +2512,92 @@ CHAKRA_API JsHasException(_Out_ bool *hasException)
     return JsNoError;
 }
 
+CHAKRA_API JsGetAndClearExceptionWithMetadata(_Out_ JsValueRef *metadata)
```

Comment:
did you consider using ContextAPIWrapper, it does bunch of checks already which you are performing in this function?

---
In reply to: [115546477](https://github.com/Microsoft/ChakraCore/pull/2936#discussion_r115546477) [](ancestors = 115546477)